src/typechecker/FStar.TypeChecker.Util.fs

```diff
@@ -1975,7 +1975,7 @@ let rec check_erased (env:Env.env) (t:term) : isErased =
      *       cases like the int types in FStar.Integers
      *     So we iterate over all the branches and return a No if possible
      *)
-    | Tm_match (_, _, branches), _ ->
+    | Tm_match (_, _, branches, _), _ ->  //AR: TODO: use rcopt if present
```

Comment:
Can we add that? or else this TODO will linger

Reply:
This TODO was misleading, sorry about this, I have removed it now. This function is checking whether the type passed to it is an erased type or not. So we actually have to inspect the type. If the type is a match node, its residual comp won't help since that would be one level up.